File: Pandacap.LowLevel/Data/Credentials/RedditCredentials.fs
namespace Pandacap.Data

open System.ComponentModel.DataAnnotations
open Pandacap.PlatformBadges

/// The active credentials for the Reddit API.
type RedditCredentials() =
    [<Key>]
    member val Username = "" with get, set

    member val AccessToken = "" with get, set
    member val RefreshToken = "" with get, set

    interface IExternalCredentials with
        member this.Username = this.Username
        member _.Platform = Reddit

File: Pandacap.Database/RedditCredentials.cs
using Pandacap.UI.Badges;
using Pandacap.UI.Elements;
using System.ComponentModel.DataAnnotations;

namespace Pandacap.Database
{
    public class RedditCredentials : IExternalCredentials
    {
        [Key]
        public string Username { get; set; } = "";

        public string AccessToken { get; set; } = "";
        public string RefreshToken { get; set; } = "";

        Badge IExternalCredentials.Badge =>
            Badges.Reddit.WithText(Username);
    }
}
